src/Compilers/Core/RebuildTest/RebuildCommandLineTests.cs

```diff
@@ -227,23 +274,43 @@ Imports System
         Console.WriteLine(CStr(True))
     End Sub
 End Module
+");
+
+            AddSourceFile("lib1.vb", @"
+Imports System
+Public Module M
+    Public Function Add(left As Integer, right As Integer) As Integer
+        return left + right
+    End Function
+End Module
 ");
         }
 
         public static IEnumerable<object?[]> GetVisualBasicData()
         {
             var list = new List<object?[]>();
 
-            Add(Permutate(new CommandInfo("hw.vb /target:exe /debug:embedded", "test.exe", null)));
+            Permutate(
+                new CommandInfo("hw.vb /debug:embedded", "test.exe", null),
+                PermutateOptimizations, PermutateRuntime, PermutateExeKinds);
+            Permutate(
+                new CommandInfo("lib1.vb /target:library /debug:embedded", "test.dll", null),
+                PermutateOptimizations, PermutateRuntime);
+            Permutate(
+                new CommandInfo(@"lib1.vb /debug:embedded /d:_MYTYPE=""Empty"" /vbruntime:Microsoft.VisualBasic.dll", "test.dll", null),
+                PermutateOptimizations, PermutateDllKinds);
 
             return list;
 
-            IEnumerable<CommandInfo> Permutate(CommandInfo commandInfo)
+            void Permutate(CommandInfo commandInfo, params Func<CommandInfo, IEnumerable<CommandInfo>>[] permutations)
```

Comment:
I am a little nervous that we are going to find out about n-factorial eventually

Reply:
Not gotta lie, I screwed this up once and was pleasantly surprised at how well test explorer scaled to my massive mistake